### ISSUE TITLE
Xorg with root thru vnc does not connect

### ISSUE THREAD

**Issue by robertwenquan** [codemaze-fun/noi-linux#32]
with the Xorg auth, 
it does not auth thru root

![image](https://user-images.githubusercontent.com/8893623/223031177-2651805f-dcae-4a99-b9e4-0e2249f23dbe.png)

hrowing errors
```
The VNC desktop is:      b50cfb64be48:0
PORT=5900
xrdp[10]: (10)(140573296506688)[INFO ] starting xrdp with pid 10

xrdp[10]: (10)(140573296506688)[INFO ] address [0.0.0.0] port [3389] mode 1

xrdp[10]: (10)(140573296506688)[INFO ] listening to port 3389 on 0.0.0.0

xrdp[10]: (10)(140573296506688)[INFO ] xrdp_listen_pp done

xrdp[10]: (10)(140573296506688)[INFO ] Socket 11: AF_INET6 connection received from ::ffff:172.17.0.1 port 44078

xrdp[10]: (10)(140573296506688)[DEBUG] Closed socket 11 (AF_INET6 ::ffff:172.17.0.2 port 3389)

xrdp[14]: (14)(140573296506688)[DEBUG] Closed socket 10 (AF_INET6 :: port 3389)

xrdp[14]: (14)(140573296506688)[INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem

xrdp[14]: (14)(140573296506688)[INFO ] Using default X.509 key file: /etc/xrdp/key.pem

xrdp[14]: (14)(140573296506688)[DEBUG] TLSv1.3 enabled

xrdp[14]: (14)(140573296506688)[DEBUG] TLSv1.2 enabled

xrdp[14]: (14)(140573296506688)[DEBUG] Security layer: requested 11, selected 1

xrdp[14]: (14)(140573296506688)[DEBUG] Closed socket 11 (AF_INET6 ::ffff:172.17.0.2 port 3389)

xrdp[10]: (10)(140573296506688)[INFO ] Socket 11: AF_INET6 connection received from ::ffff:172.17.0.1 port 44080

xrdp[10]: (10)(140573296506688)[DEBUG] Closed socket 11 (AF_INET6 ::ffff:172.17.0.2 port 3389)

xrdp[15]: (15)(140573296506688)[DEBUG] Closed socket 10 (AF_INET6 :: port 3389)

xrdp[15]: (15)(140573296506688)[INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem

xrdp[15]: (15)(140573296506688)[INFO ] Using default X.509 key file: /etc/xrdp/key.pem

xrdp[15]: (15)(140573296506688)[DEBUG] TLSv1.3 enabled

xrdp[15]: (15)(140573296506688)[DEBUG] TLSv1.2 enabled

xrdp[15]: (15)(140573296506688)[DEBUG] Security layer: requested 11, selected 1

xrdp[15]: (15)(140573296506688)[INFO ] connected client computer name: ROBERT-MATEBK-X

xrdp[15]: (15)(140573296506688)[INFO ] adding channel item name rdpdr chan_id 1004 flags 0x80800000

xrdp[15]: (15)(140573296506688)[INFO ] adding channel item name rdpsnd chan_id 1005 flags 0xc0000000

xrdp[15]: (15)(140573296506688)[INFO ] adding channel item name cliprdr chan_id 1006 flags 0xc0a00000

xrdp[15]: (15)(140573296506688)[INFO ] adding channel item name drdynvc chan_id 1007 flags 0xc0800000

xrdp[15]: (15)(140573296506688)[INFO ] TLS connection established from ::ffff:172.17.0.1 port 44080: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_0000000f_wm_login_mode_event_00000001

xrdp[15]: (15)(140573296506688)[INFO ] Cannot find keymap file /etc/xrdp/km-00000804.ini

xrdp[15]: (15)(140573296506688)[INFO ] Cannot find keymap file /etc/xrdp/km-00000804.ini

xrdp[15]: (15)(140573296506688)[INFO ] Loading keymap file /etc/xrdp/km-00000409.ini

xrdp[15]: (15)(140573296506688)[WARN ] local keymap file for 0x00000804 found and doesn't match built in keymap, using local keymap file

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350

xrdp-sesman[12]: (12)(140159498221120)[INFO ] A connection received from ::ffff:127.0.0.1 port 59408

xrdp[15]: (15)(140573296506688)[INFO ] xrdp_wm_log_msg: sesman connect ok

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_wm_log_msg: sending login info to session manager, please wait...

xrdp[15]: (15)(140573296506688)[DEBUG] return value from xrdp_mm_connect 0

xrdp-sesman[12]: (12)(140159498221120)[INFO ] ++ created session (access granted): username root, ip ::ffff:172.17.0.1:44080 - socket: 11

xrdp-sesman[12]: (12)(140159498221120)[INFO ] starting Xorg session...

xrdp-sesman[12]: (12)(140159498221120)[DEBUG] Closed socket 8 (AF_INET6 :: port 5910)

xrdp-sesman[12]: (12)(140159498221120)[DEBUG] Closed socket 8 (AF_INET6 :: port 6010)

xrdp-sesman[12]: (12)(140159498221120)[DEBUG] Closed socket 8 (AF_INET6 :: port 6210)

xrdp-sesman[16]: (16)(140159498221120)[INFO ] calling auth_start_session from pid 16

xrdp-sesman[12]: (12)(140159498221120)[DEBUG] Closed socket 7 (AF_INET6 ::ffff:127.0.0.1 port 3350)

xrdp[15]: (15)(140573296506688)[INFO ] xrdp_wm_log_msg: login successful for display 10

xrdp-sesman[16]: pam_unix(xrdp-sesman:session): session opened for user root by (uid=0)
xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_wm_log_msg: started connecting

xrdp-sesman[16]: pam_systemd(xrdp-sesman:session): Failed to connect to system bus: No such file or directory
xrdp-sesman[16]: (16)(140159498221120)[DEBUG] Closed socket 6 (AF_INET6 ::ffff:127.0.0.1 port 3350)

xrdp-sesman[16]: (16)(140159498221120)[DEBUG] Closed socket 7 (AF_INET6 ::ffff:127.0.0.1 port 3350)

xrdp-sesman[18]: (18)(140159498221120)[INFO ] /usr/lib/xorg/Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log

xrdp[15]: (15)(140573296506688)[INFO ] lib_mod_log_peer: xrdp_pid=15 connected to X11rdp_pid=18 X11rdp_uid=0 X11rdp_gid=0 client_ip=::ffff:172.17.0.1 client_port=44080

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_wm_log_msg: connected ok

xrdp-sesman[16]: (16)(140159498221120)[CORE ] waiting for window manager (pid 17) to exit

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_mm_connect_chansrv: chansrv connect successful

xrdp[15]: (15)(140573296506688)[DEBUG] Closed socket 17 (AF_INET6 ::ffff:127.0.0.1 port 59408)

xrdp-sesman[16]: (16)(140159498221120)[CORE ] window manager (pid 17) did exit, cleaning up session

xrdp-sesman[16]: (16)(140159498221120)[INFO ] calling auth_stop_session and auth_end from pid 16

xrdp-sesman[16]: pam_unix(xrdp-sesman:session): session closed for user root
xrdp-sesman[16]: (16)(140159498221120)[DEBUG] cleanup_sockets:

xrdp[15]: (15)(140573296506688)[DEBUG] Closed socket 11 (AF_INET6 ::ffff:172.17.0.2 port 3389)

xrdp-sesman[16]: (16)(140159498221120)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdp_chansrv_audio_out_socket_10

xrdp[15]: (15)(140573296506688)[DEBUG] xrdp_mm_module_cleanup

xrdp-sesman[16]: (16)(140159498221120)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdp_chansrv_audio_in_socket_10

xrdp[15]: (15)(140573296506688)[DEBUG] Closed socket 18 (AF_UNIX)

xrdp-sesman[16]: (16)(140159498221120)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdpapi_10

xrdp[15]: (15)(140573296506688)[DEBUG] Closed socket 19 (AF_UNIX)

xrdp-sesman[12]: (12)(140159498221120)[INFO ] ++ terminated session:  username root, display :10.0, session_pid 16, ip ::ffff:172.17.0.1:44080 - socket: 11

```

**Comment by robertwenquan** [comment]
knonw non-root user also doesn't auth

```
xrdp[575]: (575)(140511397566272)[INFO ] lib_mod_log_peer: xrdp_pid=575 connected to X11rdp_pid=578 X11rdp_uid=1000 X11rdp_gid=1000 client_ip=::ffff:172.17.0.1 client_port=33244

xrdp[575]: (575)(140511397566272)[DEBUG] xrdp_wm_log_msg: connected ok

xrdp-sesman[576]: (576)(140166113334848)[CORE ] waiting for window manager (pid 577) to exit

xrdp[575]: (575)(140511397566272)[DEBUG] xrdp_mm_connect_chansrv: chansrv connect successful

xrdp[575]: (575)(140511397566272)[DEBUG] Closed socket 17 (AF_INET6 ::ffff:127.0.0.1 port 44672)

xrdp-sesman[576]: (576)(140166113334848)[CORE ] window manager (pid 577) did exit, cleaning up session

xrdp-sesman[576]: (576)(140166113334848)[INFO ] calling auth_stop_session and auth_end from pid 576

xrdp-sesman[576]: pam_unix(xrdp-sesman:session): session closed for user codemaze
xrdp-sesman[576]: (576)(140166113334848)[DEBUG] cleanup_sockets:

xrdp[575]: (575)(140511397566272)[DEBUG] Closed socket 11 (AF_INET6 ::ffff:172.17.0.2 port 3389)

xrdp-sesman[576]: (576)(140166113334848)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdp_chansrv_audio_out_socket_10

xrdp[575]: (575)(140511397566272)[DEBUG] xrdp_mm_module_cleanup

xrdp-sesman[576]: (576)(140166113334848)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdp_chansrv_audio_in_socket_10

xrdp[575]: (575)(140511397566272)[DEBUG] Closed socket 18 (AF_UNIX)

xrdp-sesman[576]: (576)(140166113334848)[DEBUG] cleanup_sockets: deleting /run/xrdp/sockdir/xrdpapi_10

xrdp[575]: (575)(140511397566272)[DEBUG] Closed socket 19 (AF_UNIX)

xrdp-sesman[572]: (572)(140166113334848)[INFO ] ++ terminated session:  username codemaze, display :10.0, session_pid 576, ip ::ffff:172.17.0.1:33244 - socket: 11
```

**Comment by robertwenquan** [comment]
fixed in https://github.com/codemaze-fun/noi-linux/pull/40
